src/search_algo.hpp

```diff
@@ -595,8 +655,11 @@ search(LocalDataHolder<TGlobalHolder> & lH)
                     continue;
 
                 // locate hits
-                for (auto [ subjNo, subjOffset ] : cursor.lazy_locate())
-                {
+                for (auto [subjNo, subjOffset] : fmindex_collection::LocateLinear{lH.gH.indexFile.index, cursor}) {
+                    // !TODO Should this be handled by the cursor?
+                    if (c_indexType == DbIndexType::FM_INDEX) {
+                        subjOffset -= seedLength;
```

Comment:
The index returns the end of the match?

Reply:
Yes, since we are using the reversed text to be able to extend our query to the right, the suffix array reports the ending of our text, not the beginning. To be able to determine the start position the cursor would need to carry the information how often it was extended....
But, I think I should change this. Just haven't got to it yet and seemed to have a low priority.